syntax = "proto3";

package producto;

// El servicio que define las funciones remotas
service ProductoService {
  rpc IngresarProducto(ProductoRequest) returns (ProductoResponse) {}
}

// El mensaje que el cliente env√≠a
message ProductoRequest {
  string nombre = 1;
  string descripcion = 2;
  float precio = 3;
  int32 stock_inicial = 4;
  bytes foto = 5; // Para los bytes de la imagen
}

// La respuesta que el servidor devuelve
message ProductoResponse {
  bool exito = 1;
  string mensaje = 2;
}